predicators/approaches/grammar_search_invention_approach.py

```diff
@@ -150,6 +164,48 @@ def pretty_str(self) -> Tuple[str, str]:
         return vars_str, body_str
 
 
+@dataclass(frozen=True, eq=False, repr=False)
+class _DoubleAttributeCompareClassifier(_BinaryClassifier):
+    """Compare a single feature value with a constant value."""
+    object1_index: int
+    object1_type: Type
+    attribute1_name: str
+    object2_index: int
+    object2_type: Type
+    attribute2_name: str
+    constant: float
+    constant_idx: int
+    compare: Callable[[float, float], bool]
+    compare_str: str
+
+    def _classify_object(self, s: State, obj1: Object, obj2: Object) -> bool:
+        assert obj1.type == self.object1_type
+        assert obj2.type == self.object2_type
+        return self.compare(
+            abs(
+                s.get(obj1, self.attribute1_name) -
```

Comment:
I wonder if we should name things in terms of "difference" instead of "double", since "double" suggests that we might compare two features in a more general way